syntax = "proto3";

package proto_catalogue;

import "map_renderer.proto";
import "transport_router.proto";
import "graph.proto";

message Coordinates {
    double lat = 1;
    double lng = 2;
}

message Stop {
    string stop_name = 1;
    Coordinates coordinates = 2;
    repeated string buses_on_stop = 3;
}

message Bus {
    string bus_number = 1;
    repeated string stops = 2;
    bool roundtrip = 3;
}

message BusInfo {
    int32 stops_count = 1;
    int32 unique_stops = 2;
    double geo_route_length = 3;
    int32 m_route_length = 4;
    double curvature = 5;
}

message Distances {
    string from = 1;
    string to = 2;
    int32 distance = 3;
}

message TransportCatalogue {
    repeated Bus buses = 1;
    repeated Stop stops = 2;
    repeated Distances stop_distances = 3;
    proto_map_renderer.RenderSettings render_settings = 4;
    proto_transport_router.TransportRouter router = 5;
    proto_graph.Graph graph = 6;
}